database/dbService.js

```diff
@@ -10,24 +10,34 @@ module.exports = {
             var sql = queries.insertUser;
             con.query(sql, [user.id, user.gender, user.name.givenName, user.name.familyName, user.emails[0].value, user._json.placesLived[0].value, user.phoneNumber]
             , function (err, result) {
-                con.release();                
                 if (err) throw err;
                 callback(result.insertId);
+                con.release();                                
+            });
+        });
+    },
+
+    insertActivityLog: function(ID, firstName, lastName, action) {
+        this.tryConnect().getConnection(function (err, con) {
+            var sql = queries.insertActivityLog;
+            con.query(sql, [ID, firstName, lastName, action], function (err, result) {
+                if (err) throw err;
+                con.release();
             });
         });
     },
 
     replaceUserOnDuplicate: function(user, callback) {
         this.tryConnect().getConnection(function(err, con) {
-            var sql = queries.ReplaceUserOnDuplicate;
+            var sql = queries.replaceUserOnDuplicate;
             // Insert parameters
             con.query(sql, [user.id, user.googleID, user.gender, user.firstName, user.lastName, user.email, user.isProfileSetUp, user.location, user.phoneNumber,
                  // On Duplicate Key Update parameters
                  user.googleID, user.gender, user.firstName, user.lastName, user.email, user.isProfileSetUp, user.location, user.phoneNumber], 
                  function (err, result) {
-                con.release();
                 if (err) throw err;
-                return callback(result.insertId);
+                callback(result.insertId);
+                con.release();                
             });
```

Comment:
Realized we should release the connection back to the pool at the end of the query